experimental/JuLie/tableaux.jl

```diff
@@ -0,0 +1,717 @@
+    standard_tableaux(s::Vector{Integer})
+
+Returns a list of all standard tableaux of a given shape.
+"""
+function standard_tableaux(s::Partition)
+  Tabs = Vector{Tableau}()
+  if isempty(s)
+    push!(Tabs, Tableau(Vector{Int}[]))
+    return Tabs
+  end
+  n_max = sum(s)
+  ls = length(s)
+
+  Tab = Tableau([ [0 for j = 1:s[i]] for i = 1:length(s)])
+  sub_s = [0 for i=1:length(s)]
+  Tab[1][1] = 1
+  sub_s[1] = 1
+  tracker_row = [0 for i=1:n_max]
+  tracker_row[1] = 1
+
+  n = 1
+  i = 1
+  j = 2
+
+  while n > 0
+    if n == n_max || i > ls
+      if n == n_max
+        push!(Tabs,Tableau([copy(row) for row in Tab]))
+      end
+      Tab[tracker_row[n]][sub_s[tracker_row[n]]] = 0
+      i = tracker_row[n] + 1
+      if i <= ls
+        j = sub_s[i] + 1
+      end
+      n -= 1
+      sub_s[i-1] -= 1
+    elseif j<=s[i] && (i==1 || j<=sub_s[i-1])
+      n += 1
+      Tab[i][j] = n
+      sub_s[i] += 1
+      tracker_row[n] = i
+      i = 1
+      j = sub_s[1] + 1
+    else
+      i += 1
+      if i <= ls
+        j = sub_s[i] + 1
+      end
+    end
+  end
+
+  return Tabs
+end
+
+
+function standard_tableaux(s::Vector{T}) where T<:Integer
+  return standard_tableaux(Partition(s))
+end
+
+
+@Markdown.doc """
+    standard_tableaux(n::Integer)
+
+Returns a list of all standard tableaux with n boxes.
+"""
+function standard_tableaux(n::Integer)
+  n>=0 || throw(ArgumentError("n ≥ 0 required"))
+  ST = Vector{Tableau}()
+  for s in partitions(n)
+    append!(ST, standard_tableaux(s))
+  end
+  return ST
+end
+
+
+
+@Markdown.doc """
+    hook_length(lambda::Partition, i::Integer, j::Integer)
+
+Consider the Young diagram of a partition ``λ``. The **hook length** of a
+box, is the number of boxes to the right in the same row + the number
+of boxes below in the same column + 1. The function returns the hook
+length of the box with coordinates `(i,j)`. The functions assumes that
+the box exists.
+"""
+function hook_length(lambda::Partition, i::Integer, j::Integer)
+  h = lambda[i] - j + 1
+  k = i + 1
+  while k<=length(lambda) && lambda[k]>=j
+    k += 1
+    h += 1
+  end
+  return h
+end
+
+@Markdown.doc """
+    hook_length(Tab::Tableau, i::Integer, j::Integer)
+
+Shortcut for ```hook_length(shape(Tab), i, j)```.
+"""
+function hook_length(Tab::Tableau, i::Integer, j::Integer)
+  return hook_length(shape(Tab),i,j)
+end
+
+
+@Markdown.doc """
+    hook_lengths(lambda::Partition)
+
+Returns the tableau of shape ``λ`` in which the entry at position `(i,j)`
+is equal to the hook length of the corresponding box.
+"""
+function hook_lengths(lambda::Partition)
+  if isempty(lambda)
+    return Tableau(Vector{Int}[])
+  end
+  Tab = [ [hook_length(lambda,i,j) for j in 1:lambda[i]] for i in 1:length(lambda) ]
+  return Tableau(Tab)
+end
+
+
+
+
+@doc Markdown.doc"""
+    num_standard_tableaux(lambda::Partition)
+
+Returns the number $f^λ$ of standard tableaux of shape ``λ`` using the hook length formula
+
+$f^λ = \frac{n!}{\prod_{i,j} h_λ(i,j)},$
```

Comment:
for a stand-alone formula, maybe use display math, i.e.:
```suggestion
$$f^λ = \frac{n!}{\prod_{i,j} h_λ(i,j)},$$
```